src/next/main.ts

```diff
@@ -0,0 +1,143 @@
+ * @param cb The callback function for the command.
+ * @returns A new callback with telemetry logic injected.
+ */
+export function getWrappedCommand(commandName: string, cb: (...args: any[]) => any): (...args: any[]) => any {
+    return async (...args: any[]) => {
+        let error = undefined;
+        let oId = uuid.v4();
+        const startAt: number = Date.now();
+        try {
+            sendOpStart(oId, commandName);
+            return await cb(...args);
+        } catch (e) {
+            error = e;
+            sendError(oId, error);
+        } finally {
+            const duration = Date.now() - startAt;
+            sendOpEnd(oId, commandName, duration, error);
+        }
+    }
+};
+
+/**
+ * Send ERROR telemetry data, supposed to be used when an error occurs.
+ * @param oId Operation Id.
+ * @param errorObject An object containing error details.
+ */
+export function sendError(oId: string, errorObject: any) {
+    if (!errorObject || errorObject.errorCode === ErrorCode.NO_ERROR) {
+        if (_isDebug) {
+            console.warn("No error indicated by: ", errorObject);
+        }
+    } else {
+        const errorProperties: Properties = getErrorProperties(errorObject);
+        report(EventName.ERROR, { oId, ...errorProperties });
+    }
+}
+
+/**
+ * Send OPERATION_START telemetry data, supposed to be used when an operation starts.
+ * @param oId Operation Id.
+ * @param oName Operation name.
+ */
+export function sendOpStart(oId: string, oName: string) {
+    report(EventName.OPERATION_START, { oId, oName });
+}
+
+/**
+ * Send OPERATION_END telemetry data, supposed to be used when an operation ends.
+ * @param oId Operation Id.
+ * @param oName Operation name.
+ * @param duration Time elapsed for the operation, in milliseconds.
+ * @param errorObject An optional object containing error details.
+ */
+export function sendOpEnd(oId: string, oName: string, duration: number, errorObject?: any) {
+    const errorProperties: Properties = getErrorProperties(errorObject);
+    report(EventName.OPERATION_END, { oId, oName, ...errorProperties }, { duration });
+}
+
+
+type Properties = {
+    [key: string]: string;
+};
+
+type Measurements = {
+    [key: string]: number;
+};
+
+function report(eventName: string, properties?: Properties, measurements?: Measurements): void {
+    if (reporter) {
+        reporter.sendTelemetryEvent(eventName, properties, measurements);
+        if (_isDebug) {
+            console.log(eventName, { eventName, properties, measurements });
+        }
+    } else {
+        if (_isDebug) {
+            console.warn("TelemetryReporter is not initialized.");
+        }
+    }
+}
+
+const ALLOWED_KEYS_FOR_ERROR: string[] = [
+    "errorCode",// Preserve 0 for no error, 1 for general error.
+    "name",     // For name of an Error.
+    "message",  // For message of an Error.
+    "stack",    // For callstack of an Error.
+];
+
+function getErrorProperties(object: any): Properties {

```

Comment:
`extract` is more vivid to describe the behavior while `get` sounds too lightweight.

To me, get means `get without further processing`, like a getter of a property.